File: CodeGenDotNet 159059-1372 (a)/CSharp/Chapter 3/BruteForceExample/Support.cs
using System;

namespace KADGen.BruteForceExample
{
	/// <summary>
	/// Simple Utility files supporting code generation
	/// </summary>
	public class Support
	{
		public const string DQ = "\"";

		public static void FileOpen(	System.CodeDom.Compiler.IndentedTextWriter inWriter,
										string import,
										string fileName,
										string genDateTime )
		{
			inWriter.WriteLine( "using System;" );
			foreach( string s in import.Split( ',' ) )
			{
				inWriter.WriteLine( "using " + s.Trim() + ";" );
			}
			
			inWriter.WriteLine( "/// <summary>" );
			inWriter.WriteLine( "/// " + System.IO.Path.GetFileName( fileName ) );
			inWriter.WriteLine( "/// Last Generated on Date: " + genDateTime );
			inWriter.WriteLine( "/// </summary>" );
		}

		public static void MakeRegion( System.CodeDom.Compiler.IndentedTextWriter inWriter, string regionName )
		{
			inWriter.WriteLine( "#region " + regionName );
		}

		public static void EndRegion( System.CodeDom.Compiler.IndentedTextWriter inWriter )
		{
			inWriter.WriteLine( "#endregion" );
		}

		public static void WriteLineAndIndent( System.CodeDom.Compiler.IndentedTextWriter inWriter, string text )
		{
			inWriter.WriteLine( text );
			inWriter.Indent ++;
		}

		public static void WriteLineAndOutdent( System.CodeDom.Compiler.IndentedTextWriter inWriter, string text )
		{
			inWriter.Indent --;
			inWriter.WriteLine( text );
		}
	}
}

File: CodeGenDotNet 159059-1372 (a)/VBCode/Chapter 3/BruteForceExample/Support.vb
Option Strict On
Option Explicit On 

Imports System

#Region "Description"
'Simple Utility files supporting code generation
#End Region

Public Class Support
   Public Const DQ As String = """"

   Public Shared Sub FileOpen( _
                  ByVal inWriter As CodeDom.Compiler.IndentedTextWriter, _
                  ByVal import As String, _
                  ByVal filename As String, _
                  ByVal gendatetime As String)
      inWriter.WriteLine("Option Strict On")
      inWriter.WriteLine("Option Explicit On")
      inWriter.WriteLine("")
      inWriter.WriteLine("Imports System")
      For Each s As String In import.Split(","c)
         inWriter.WriteLine("Imports " & s)
      Next
      MakeRegion(inWriter, "Description")
      inWriter.WriteLine("'")
      inWriter.WriteLine("'" & IO.Path.GetFileName(filename))
      inWriter.WriteLine("' Last Genned on Date: " & gendatetime)
      inWriter.WriteLine("'")
      inWriter.WriteLine("'")
      EndRegion(inWriter)
   End Sub

   Public Shared Sub MakeRegion( _
                  ByVal inWriter As CodeDom.Compiler.IndentedTextWriter, _
                  ByVal regionName As String)
      inWriter.WriteLine("")
      inWriter.WriteLineNoTabs("#Region " & DQ & regionName & DQ)
   End Sub

   Public Shared Function EndRegion(ByVal inwriter As CodeDom.Compiler.IndentedTextWriter) As String
      inwriter.WriteLineNoTabs("#End Region")
   End Function

   Public Shared Sub WriteLineAndIndent(ByVal inwriter As CodeDom.Compiler.IndentedTextWriter, ByVal text As String)
      inwriter.WriteLine(text)
      inwriter.Indent += 1
   End Sub

   Public Shared Sub WriteLineAndOutdent(ByVal inwriter As CodeDom.Compiler.IndentedTextWriter, ByVal text As String)
      inwriter.Indent -= 1
      inwriter.WriteLine(text)
   End Sub

End Class
